qcodes/dataset/measurements.py

```diff
@@ -556,7 +556,7 @@ def __enter__(self) -> DataSaver:
         if self._dataset_class is DataSet:
             dataset_class = cast(Type[DataSet], self._dataset_class)
             self.ds = dataset_class(
-                name=self.name,
+                name='results',
```

Comment:
I don't think this is the level at which we want to change this. We still want to pass the name to dataset so that it can be used in the name column of the runs table. 

What we want to do is to change `_insert_run` so that `formatted_name` does not contain the measurement name but is just of a default format of `results-expid-counter`

Reply:
![Screenshot 2021-07-08 121533](https://user-images.githubusercontent.com/548266/124905343-3cc5cc00-dfe6-11eb-9900-c1b1ef847312.png)

As in this example the name column should stay the same but the result_table_name should change

Reply:
Ok, I'll update it